python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+because grouping two computations together will run twice as fast!) by
+creating a tuple or struct:
+
+    >>> mt.aggregate_entries((agg.stats(mt.DP), agg.stats(mt.GQ)))
+    (Struct(mean=41.83915800445897, stdev=41.93057654787303, min=0.0, max=450.0, n=34537, sum=1444998.9999999995),
+    Struct(mean=67.73196915777027, stdev=29.80840934057741, min=0.0, max=99.0, n=33720, sum=2283922.0000000135))
+
+See the :ref:`sec-aggregators` page for the complete list of aggregator
+functions.
+
+Group-By
+========
+
+Matrix tables can be aggregated along the row or column axis to produce a new
+matrix table.
+
+- :meth:`.MatrixTable.group_rows_by`
+- :meth:`.MatrixTable.group_cols_by`
+
+First let's add a random phenotype as a new column field `case_status` and then
+compute statistics about the entry field `GQ` for each grouping of `case_status`.
+
+    >>> mt_ann = mt.annotate_cols(case_status = hl.cond(hl.rand_bool(0.5),
+    ...                                                 "CASE",
+    ...                                                 "CONTROL"))
+
+Next we group the columns by `case_status` and aggregate:
+
+    >>> mt_grouped = (mt_ann.group_cols_by(mt_ann.case_status)
+    ...                 .aggregate(gq_stats = agg.stats(mt_ann.GQ)))
+
+    >>> print(mt_grouped.entry.dtype.pretty())
+    struct {
+        gq_stats: struct {
+            mean: float64,
+            stdev: float64,
+            min: float64,
+            max: float64,
+            n: int64,
+            sum: float64
+        }
+    }
+
+    >>> print(mt_grouped.col.dtype)
+    struct{status: str}
+
+Joins
+=====
+
+Joins on two-dimensional data are significantly more complicated than joins
+in one dimension, and Hail does not yet support the full range of
+joins on both dimensions of a matrix table.
+
+:class:`.MatrixTable` has methods for concatenating rows or columns:
+
+- :meth:`.MatrixTable.union_cols`
+- :meth:`.MatrixTable.union_rows`
+
+:meth:`.MatrixTable.union_cols` joins matrix tables together by performing an
+inner join on rows while concatenating columns together (similar to `paste` in
+Unix). Likewise, :meth:`.MatrixTable.union_rows` performs an inner join on
+columns while concatenating rows together (similar to `cat` in Unix).
+
+In addition, Hail provides support for joining data from multiple sources together
+if the keys of each source are compatible (same order and type, but the names do
+not need to be identical) using Python's bracket notation ``[]``. The arguments
+inside the brackets are the destination key as a single value or a tuple if there
+are multiple destination keys.
+
+For example, we can annotate rows with row fields from another matrix table or
+table. Let `gnomad_data` be a :class:`.Table` keyed by two row fields with type
+``locus`` and ``array<str>``, which matches the row keys of `mt`:
+
+    >>> mt_new = mt.annotate_rows(gnomad_ann = gnomad_data[mt.locus, mt.alleles])
```

Comment:
I think I'd start with the example, the paragraph before last is a bit hard to parse.

Reply:
thought the previous paragraph works well. Maybe its just the one before last that's hard to parse